Lib/python/std_filesystem.i

```diff
@@ -0,0 +1,85 @@
+/*
+  filesystem::path
+*/
+%{
+#if __cplusplus >= 201703L
+#include <filesystem>
+#include <Python.h>
+#endif // __cplusplus >= 201703L
+%}
+
+%fragment("common", "header") {
+    PyObject * importPathCls() {
+        PyObject * module = PyImport_ImportModule("pathlib");
+        PyObject * cls = PyObject_GetAttrString(module, "Path");
+        Py_DECREF(module);
+        return cls;
+    }
+
+    bool isPathInstance(PyObject * obj) {
+        PyObject * cls = importPathCls();
+        bool is_instance =  PyObject_IsInstance(obj, cls);
+        Py_DECREF(cls);
+        return is_instance;
+    }
+
+    const char * pathToStr(PyObject * obj) {
+        PyObject * str_obj = PyObject_Str(obj);     // New reference
+        const char * s = PyUnicode_AsUTF8(str_obj); // This stores the UTF-8 representation buffer within str_obj
+        // Py_DECREF(str_obj);                      // So we don't decref here
```

Comment:
So does this mean we end up leaking `str_obj`?

Maybe this helper would be better just hand-inlined at each of the two uses, or else split into two variant helpers which respectively construct and return a `std::filesystem::path` and `std::filesystem::path*` and so can do the `Py_DECREF()`.